packages/monitors/src/exchange-rates.ts

```diff
@@ -0,0 +1,104 @@
+import { getProviders } from './providers';
+import { ethers } from 'ethers';
+import { IAggregatorABI } from '@notional-finance/contracts';
+import { aggregate, AggregateCall } from '@notional-finance/multicall';
+import { log } from '@notional-finance/logging';
+import { AggregateCallList, JobOptions, MonitorJob } from './types';
+
+let providers: Record<string, ethers.providers.JsonRpcBatchProvider> = {};
+
+async function aggregateCalls() {
+  const resp = await Promise.all(
+    [...getCallMap().values()].map(async ({ provider, calls }) => {
+      const { name } = await provider.getNetwork();
+      if (calls.length === 0)
+        return { network: name, blockNumber: 0, results: {} };
+      const { blockNumber, results } = await aggregate(calls, provider);
+      return { network: name, blockNumber, results };
+    })
+  );
+  return resp
+    .filter((data) => data.blockNumber > 0)
+    .reduce(
+      (prev, { network, blockNumber, results }) => ({
+        ...prev,
+        [network]: { blockNumber, results },
+      }),
+      {}
+    );
+}
+
+function getCallMap(): Map<number, AggregateCallList> {
+  const ethUsdContract = new ethers.Contract(
+    '0x5f4ec3df9cbd43714fe2740f5e3616155c5b8419',
+    IAggregatorABI,
+    providers['mainnet']
+  );
+  const btcUsdContract = new ethers.Contract(
+    '0xf4030086522a5beea4988f8ca5b36dbc97bee88c',
+    IAggregatorABI,
+    providers['mainnet']
+  );
+  const mainnetCalls: AggregateCall[] = [
+    {
+      target: ethUsdContract,
+      method: 'latestAnswer',
+      args: [],
+      key: 'eth_usd',
+      transform: (
+        r: Awaited<ReturnType<typeof ethUsdContract.latestAnswer>>
+      ) => {
+        return r.toString();
+      },
+    },
+    {
+      target: btcUsdContract,
+      method: 'latestAnswer',
+      args: [],
+      key: 'btc_usd',
+      transform: (
+        r: Awaited<ReturnType<typeof btcUsdContract.latestAnswer>>
+      ) => {
+        return r.toString();
+      },
+    },
+  ];
+
+  const goerliCalls: AggregateCall[] = [];
+
+  const callMap = new Map<number, AggregateCallList>([
+    [1, { provider: providers['mainnet'], calls: mainnetCalls }],
+    [5, { provider: providers['goerli'], calls: goerliCalls }],
+  ]);
+
+  return callMap;
+}
+
+async function run(opts: JobOptions): Promise<void> {
```

Comment:
Do whatever you'd like in the `run` method, you have access to the environment and context of the worker.
In this case, i'm just logging to datadog.